src/CoreBundle/Bridge/Chameleon/Module/NavigationTreeSingleSelect/NavigationTreeSingleSelect.php

```diff
@@ -0,0 +1,173 @@
+
+    /**
+     * {@inheritdoc}
+     */
+    public function Accept(\IMapperVisitorRestricted $visitor, $cachingEnabled, \IMapperCacheTriggerRestricted $cacheTriggerManager)
+    {
+        $fieldName = $this->inputFilterUtil->getFilteredGetInput('fieldName', '');
+        $activeNodeId = $this->inputFilterUtil->getFilteredGetInput('id', '');
+        $portalSelect = $this->inputFilterUtil->getFilteredGetInput('portalSelect', 0);
+        $isPortalSelectMode = $portalSelect === '1' ? true : false;
+
+        $this->restrictedNodes = $this->getPortalNavigationStartNodes();
+
+        $rootTreeId = $this->getPortalTreeRootNodeId();
+        if ('' !== $rootTreeId) {
+            $rootNode = new \TdbCmsTree();
+            $rootNode->SetLanguage(\TdbCmsUser::GetActiveUser()->GetCurrentEditLanguageID());
+            $rootNode->Load($rootTreeId);
+
+            $treeNodes = $this->createTreeDataModel($rootNode, $fieldName);
+            $visitor->SetMappedValue('treeNodes', $treeNodes);
+        }
+        $visitor->SetMappedValue('treePathHTML', $this->treePathHTML);
+        $visitor->SetMappedValue('activeId', $activeNodeId);
+        $visitor->SetMappedValue('fieldName', $fieldName);
+        $visitor->SetMappedValue('level', 0);
+        $visitor->SetMappedValue('isPortalSelectMode', $isPortalSelectMode);
+    }
+
+    private function getPortalTreeRootNodeId(): string
+    {
+        $portalId = $this->inputFilterUtil->getFilteredGetInput('portalID', '');
+
+        if ('' === $portalId) {
+            return \TCMSTreeNode::TREE_ROOT_ID;
+        }
+
+        $portal = new \TdbCmsPortal();
+        if (false === $portal->Load($portalId)) {
+            return \TCMSTreeNode::TREE_ROOT_ID;
+        }
+
+        return $portal->fieldMainNodeTree;
+    }
+
+    private function createTreeDataModel(\TdbCmsTree $node, string $fieldName, $path = ''): BackendTreeNodeDataModel
+    {
+        $treeNodeDataModel = $this->backendTreeNodeFactory->createTreeNodeDataModelFromTreeRecord($node);
+        $children = $node->GetChildren(true);
+
+        $path = $this->addBreadcrumbHtmlForNode($node, $fieldName, $path);
+
+        while ($child = $children->Next()) {
+            $childTreeNodeObj = $this->createTreeDataModel($child, $fieldName, $path);
+            $treeNodeDataModel->addChildren($childTreeNodeObj);
+        }
+
+        return $treeNodeDataModel;
+    }
+
+    protected function addBreadcrumbHtmlForNode(\TdbCmsTree $node, string $fieldName, string $path = ''): string
+    {
+        $path .= '<li class="breadcrumb-item">'.$node->fieldName."</li>\n";
+        $this->treePathHTML .= '<div id="'.$fieldName.'_tmp_path_'.$node->id.'" style="display:none;"><ol class="breadcrumb ml-0"><li class="breadcrumb-item"><i class="fas fa-sitemap"></i></li>'.$path.'</ol></div>'."\n";
+
+        return $path;
+    }
+
+    /**
+     * Fetches a list of all restricted nodes (of all portals).
+     * A restricted node is a startnavigation nodes of a portal.
```

Comment:
See comment above.

Reply:
See my comment above. I removed only the part "of a portal".